src/cbc_sdk/platform/policies.py

```diff
@@ -986,6 +1277,172 @@ def validate(self):
         return True
 
 
+class PolicyRuleConfig(MutableBaseModel):
+    """
+    Represents a rule configuration in the policy.
+
+    Create one of these objects, associating it with a Policy, and set its properties, then call its save() method to
+    add the rule configuration to the policy. This requires the org.policies(UPDATE) permission.
+
+    To update a PolicyRuleConfig, change the values of its property fields, then call its save() method.  This
+    requires the org.policies(UPDATE) permission.
+
+    To delete an existing PolicyRuleConfig, call its delete() method. This requires the org.policies(UPDATE) permission.
+
+    """
+    primary_key = "id"
+    swagger_meta_file = "platform/models/policy_ruleconfig.yaml"
+
+    def __init__(self, cb, parent, model_unique_id=None, initial_data=None, force_init=False, full_doc=False):
+        """
+        Initialize the PolicyRuleConfig object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            parent (Policy): The "parent" policy of this rule configuration.
+            model_unique_id (str): ID of the rule configuration.
+            initial_data (dict): Initial data used to populate the rule configuration.
+            force_init (bool): If True, forces the object to be refreshed after constructing.  Default False.
+            full_doc (bool): If True, object is considered "fully" initialized. Default False.
+        """
+        super(PolicyRuleConfig, self).__init__(cb, model_unique_id=model_unique_id, initial_data=initial_data,
+                                               force_init=force_init, full_doc=full_doc)
+        self._parent = parent
+        if model_unique_id is None:
+            self.touch(True)
+
+    @classmethod
+    def _create_rule_config(cls, cb, parent, data):
```

Comment:
No. This is the "hook" on which support for future specialized types of rule config objects, like host-based firewall, will get hung. I put it in here now because I _definitely know_ it will be needed later, so YAGNI doesn't apply here.

Reply:
I don't think I follow and understand why it needs to be separate from the constructor. Can you clarify your future design intent more

Reply:
Here's the way this function might be modified when we do host-based firewall:
```
    def _create_rule_config(cls, cb, parent, data):
        category = data.get('category', None)
        if category == 'host-based-firewall':
            return HostBasedFirewallRuleConfig(cb, parent, data.get("id", None), data, False, True)
        return PolicyRuleConfig(cb, parent, data.get("id", None), data, False, True)
```
This version looks for specific categories of information to create specific specialized subclassses of `PolicyRuleConfig`.

Reply:
This has been moved.